### ISSUE TITLE
Unable to use "&" as value for parameter in stream URI

### ISSUE THREAD

**Issue by uSpike** [badaix/snapcast#850]
**Describe the bug**
My wife has a spotify account with many ampersands (`&`) in the password.  At the root of this issue, I'd rather not ask her to change it.

I am unable to configure a stream with parameters like this:
```
source = spotify://librespot?name=Spotify&username=EMAIL&password=string&with&ampersands&devicename=Snapcast&bitrate=320&killall=false
```
since I imagine whatever library is parsing this stream doesn't know where the password ends and the `devicename` starts.  I've tried escaping `&` with `%26` but this seems to make no difference.

**Steps to Reproduce**
Start snapserver with the above config.

**Comment by badaix** [comment]
As a workaround you could run librespot in discovery mode, as discussed [here](https://github.com/badaix/snapcast/pull/181)

**Comment by uSpike** [comment]
@badaix hey thanks for the suggestion, I didn't know that feature existed.  It seems to work well for my use case.

**Comment by badaix** [comment]
@uSpike good to hear, the decoding is now fixed with commit a2168f1

**Comment by badaix** [comment]
Released in [Snapcast v0.25.0](https://github.com/badaix/snapcast/releases/tag/v0.25.0)
